### PR TITLE
Adding example for Spark jobs to configure event log storage as S3

### PR DESCRIPTION
**Pull request by akshayharyani**

### CODE REVIEW

File: examples/spark-3.1.2/crs/spark-pi-s3-logs.yaml [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+  imagePullSecrets:
+    - imagepull
+  mainClass: org.apache.spark.examples.SparkPi
+  mainApplicationFile: "local:///opt/mapr/spark/spark-3.1.2/examples/jars/spark-examples_2.12-3.1.2.0-eep-SNAPSHOT.jar"
+  restartPolicy:
+    type: Never
+  volumes:
+    - name: "spark-hs"
+      persistentVolumeClaim:
+        claimName: spark-hs-pvc
```
Comment (AlexNavara) [inline examples/spark-3.1.2/crs/spark-pi-s3-logs.yaml:50]: No need in volume and volumeMounts for this case

File: examples/spark-3.1.2/crs/spark-pi-s3-logs.yaml [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+  driver:
+    cores: 1
+    coreLimit: "1000m"
+    memory: "512m"
+    volumeMounts:
+      - mountPath: "/var/eventlog"
+        name: spark-hs
```
Comment (AlexNavara) [inline examples/spark-3.1.2/crs/spark-pi-s3-logs.yaml:57]: No need in volume and volumeMounts for this case

File: examples/spark-3.1.2/crs/spark-pi-s3-logs.yaml [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+    spark.eventLog.enabled: "true"
+    spark.hadoop.fs.s3a.endpoint: "http://16.0.14.114:9000"
+    spark.hadoop.fs.s3a.access.key: "AKIAIOSFODNN7EXAMPLE"
+    spark.hadoop.fs.s3a.secret.key: "wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY"
+    spark.hadoop.fs.s3a.path.style.access: "true"
+    spark.hadoop.fs.s3a.impl: "org.apache.hadoop.fs.s3a.S3AFileSystem"
+  type: Scala
+  sparkVersion: 3.1.2
+  mode: cluster
+  image: gcr.io/mapr-252711/spark-3.1.2:202110062359AN
```
Comment (AlexNavara) [inline examples/spark-3.1.2/crs/spark-pi-s3-logs.yaml:39]: This tag is outdated

File: examples/spark-3.1.2/crs/spark-pi-s3-logs.yaml [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+  imagePullPolicy: Always
+  imagePullSecrets:
+    - imagepull
+  mainClass: org.apache.spark.examples.SparkPi
+  mainApplicationFile: "local:///opt/mapr/spark/spark-3.1.2/examples/jars/spark-examples_2.12-3.1.2.0-eep-SNAPSHOT.jar"
```
Comment (AlexNavara) [inline examples/spark-3.1.2/crs/spark-pi-s3-logs.yaml:44]: Path to the main app jar has changed in latest images

File: examples/spark-3.1.2/crs/spark-pi-s3-logs.yaml [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+metadata:
+  name: spark-pi-mapr
+  namespace: test123
+spec:
+  sparkConf:
+    # Note: If you are executing the application as a K8 user that MapR can verify,
+    #       you do not need to specify a spark.mapr.user.secret
+    #spark.mapr.user.secret: spark-user-secret
+    # Note: You do not need to specify a spark.eventLog.dir
+    #       it will be auto-generated with the pattern "maprfs:///apps/spark/<namespace>"
```
Comment (AlexNavara) [inline examples/spark-3.1.2/crs/spark-pi-s3-logs.yaml:28]: Please, add here a commented out  section about a workaround for encrypted s3a case:
    spark.driver.extraJavaOptions: "-Dcom.amazonaws.sdk.disableCertChecking=true"
    spark.executor.extraJavaOptions: "-Dcom.amazonaws.sdk.disableCertChecking=true"
As of now there is no way to customize truststore, therefore spark apps will trust only DF objectstore. To make it work, users need to add the options above.

**Review by AlexNavara — CHANGES_REQUESTED** [review]

File: examples/spark-3.1.2/crs/spark-pi-s3-logs.yaml (hunk 1 above)
Comment (akshayharyani) [reply examples/spark-3.1.2/crs/spark-pi-s3-logs.yaml]: Removed.

File: examples/spark-3.1.2/crs/spark-pi-s3-logs.yaml (hunk 3 above)
Comment (akshayharyani) [reply examples/spark-3.1.2/crs/spark-pi-s3-logs.yaml]: Updated

File: examples/spark-3.1.2/crs/spark-pi-s3-logs.yaml (hunk 5 above)
Comment (akshayharyani) [reply examples/spark-3.1.2/crs/spark-pi-s3-logs.yaml:28]: Add it below

File: examples/spark-3.1.2/crs/spark-pi-s3-logs.yaml (hunk 4 above)
Comment (akshayharyani) [reply examples/spark-3.1.2/crs/spark-pi-s3-logs.yaml]: updated

File: examples/spark-3.1.2/crs/spark-pi-s3-logs.yaml (hunk 2 above)
Comment (akshayharyani) [reply examples/spark-3.1.2/crs/spark-pi-s3-logs.yaml]: removed

**Review by AlexNavara — APPROVED** [review]
LGTM

**Pull request merged**